### PR TITLE
Change default selectOnFocus behavior in Select

### PR DESCRIPTION
**Pull request by peterjcoffman**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- N/A Reviewed and tested on Mobile, or determined not required.
- N/A Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by lbwexler** [comment]
I like this one alot, in theory.  @haynesjm42  -- do you feel so strongly that this is always what you need, to the degree that we should change the default?   Sometimes aren't you just further drilling down?

You pretty often want it in a TextField as well, but we did not make it the default.  Don't feel strongly here....

**Comment by haynesjm42** [comment]
Just tried this out and works great. To @lbwexler's point about whether this should be on all the time, I'm not 100% sure, I could go either way.  Might make sense for this to be off by default, as it was selects with very large option lists where you rarely scroll through the list which prompted the ticket, pretty easy to know if you want this enable and the auto-selection of the text could get distracting on selects with smaller option lists.

**Comment by peterjcoffman** [comment]
@haynesjm42 If you want to enable this on a case-by-case basis, I think it would make more sense to just use the existing `selectOnFocus` prop. This ticket is changing the default. If you're happy with an opt-in behavior, that is already available.

**Comment by haynesjm42** [comment]
Ok I am going back to my original stance here, if you are setting up a select which you can filter, then you are expecting there to be enough options in the list that the user would want to be able to narrow it down by typing in the input. Same with allowing create I think. I like this as default behavior, but will leave it to others to make the final call.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,10 @@
 
 [Commit Log](https://github.com/xh/hoist-react/compare/v30.0.0...develop)
 
+### ⚙️ Technical
+
+* `Select` will now default to selecting contents on focus if in filter or creatable mode.
```
Comment (amcclain) [inline CHANGELOG.md]: This needs to get shifted up before the commit log link. Agree I think Technical is the correct category for it and that it's worth a one-liner in the changelog.

**Comment by amcclain** [comment]
I like this change and agree it's a better default for when the control is setup for the user to type in input.

It might have the positive side-effect of encouraging users to turn off filter mode for small select lists where they just want users to choose an existing option. I try to remember to do that, as I think having the interactive cursor in those small lookups is non-ideal.

File: CHANGELOG.md (hunk 1 above)
Comment (lbwexler) [reply CHANGELOG.md]: done

**Comment by lbwexler** [comment]
Thanks for this peter, and thanks for the strong advocacy john :-)

**Pull request merged**